source/common/http/conn_manager_impl.cc

```diff
@@ -2087,18 +2099,21 @@ void ConnectionManagerImpl::ActiveStream::recreateStream(
   // FilterState that we inherit, we'll end up copying this every time even though we could get
   // away with just resetting it to the HCM filter_state_.
   if (filter_state->hasDataAtOrAboveLifeSpan(StreamInfo::FilterState::LifeSpan::Request)) {
-    (*connection_manager_.streams_.begin())->filter_manager_.streamInfo().filter_state_ =
+    new_active_stream.filter_manager_.streamInfo().filter_state_ =
         std::make_shared<StreamInfo::FilterStateImpl>(
             filter_state->parent(), StreamInfo::FilterState::LifeSpan::FilterChain);
   }
 
   // Make sure that relevant information makes it from the original stream info
   // to the new one. Generally this should consist of all downstream related
   // data, and not include upstream related data.
-  (*connection_manager_.streams_.begin())
-      ->filter_manager_.streamInfo()
-      .setFromForRecreateStream(filter_manager_.streamInfo());
+  new_active_stream.filter_manager_.streamInfo().setFromForRecreateStream(
+      filter_manager_.streamInfo());
   new_stream.decodeHeaders(std::move(request_headers_), !proxy_body);
+  ENVOY_BUG(
```

Comment:
I think this is overkill - unit and e2e testing are good enough

Reply:
removed